browserApp/info_view.py

```diff
@@ -1,35 +1,152 @@
+except:
+    from PyQt5 import QtWidgets, QtCore, QtGui
 
+from os import listdir, open, O_RDWR
+from os.path import isfile, join
+from functools import partial
 
 class InfoView(QtWidgets.QWidget):
     def __init__(self, parent=None):
         super(InfoView, self).__init__(parent)
 
-        layout = QtWidgets.QHBoxLayout()
+        layout = QtWidgets.QVBoxLayout()
 
-        self.table = QtWidgets.QTreeWidget()
-        self.table.setColumnCount(2)
+        self.icons_widget = QtWidgets.QWidget()
+        self.icons_layout = QtWidgets.QGridLayout()
+
+        self.list_table = QtWidgets.QWidget()
+        self.list_layout = QtWidgets.QGridLayout()
+        self.list_table.setLayout(self.list_layout)
+        self.list_layout.setAlignment(QtCore.Qt.AlignTop)
+        self.file_name_header = QtWidgets.QLabel("File:")
+        self.size_header = QtWidgets.QLabel("Size:")
+        self.type_header = QtWidgets.QLabel("Type:")
+        self.created_header = QtWidgets.QLabel("Created:")
+        self.list_layout.addWidget(self.file_name_header, 0, 0)
+        self.list_layout.addWidget(self.size_header, 0, 1)
+        self.list_layout.addWidget(self.type_header, 0, 2)
+        self.list_layout.addWidget(self.created_header, 0, 3)
+
+        self.list_view_button = QtWidgets.QPushButton("List View")
+        self.icons_view_button = QtWidgets.QPushButton("Icon View")
+        self.top_bar_layout = QtWidgets.QHBoxLayout()
+        self.top_bar_layout.addWidget(self.list_view_button)
+        self.top_bar_layout.addWidget(self.icons_view_button)
+
+        self.list_view_button.clicked.connect(self.setup_list_view)
+        self.icons_view_button.clicked.connect(self.setup_icons_view)
+
+        self.stacked_pages = QtWidgets.QStackedWidget()
+        self.stacked_pages.addWidget(self.list_table)
+        self.stacked_pages.addWidget(self.icons_widget)
+
+        self.stacked_pages.setCurrentWidget(self.list_table)
 
         layout.addSpacing(1)
-        layout.addWidget(self.table)
+        layout.addLayout(self.top_bar_layout)
+        layout.addWidget(self.stacked_pages)
         layout.addSpacing(1)
 
+        self.column_index = 0
+        self.row_index = 0
 
         self.setLayout(layout)
 
     def populate(self, data):
-        print("Do something useful with: " + str(data))
+        # print("Do something useful with: " + str(data))
 
-        # Clear the table
-        self.table.clear()
+        # Clear the list_table
+        self.clear_layout(self.list_layout)
+        self.stacked_pages.setCurrentWidget(self.list_table)
 
-        # Update the table with name : value from the get_info() dict
+        # If item clicked on is a file, display info in list_table. Else, display buttons with file names.
         meta = data.get_info()
-        for k, v in meta.items():
-            item = QtWidgets.QTreeWidgetItem([k,v])
-            self.table.addTopLevelItem(item)
+        if 'type' in meta:
```

Comment:
it may be neater to write:
if 'type' not in meta:
   return
then you don't need to indent?